src/glcontext_egl.cpp

```diff
@@ -109,7 +109,16 @@ EGL_IMPORT
 		{
             EGLSurface defaultSurface = eglGetCurrentSurface(EGL_DRAW);
 
-			m_surface = eglCreateWindowSurface(m_display, _config, _nwh, NULL);
+ 			if(NULL == _nwh)
+			{
+				// Create an EGL pbuffer surface
+				m_surface = eglCreatePbufferSurface(m_display, _config, NULL);
```

Comment:
You didn't import this function for `BGFX_USE_GL_DYNAMIC_LIB` case.

Reply:
Also ideally if we can skip any surface, since it never will be visible, nor can be accessed in any other way, it would be great to skip it.

If skipping is not possible it should be minimal dimensions.

Reply:
If the [EGL_KHR_surfaceless_context]( https://registry.khronos.org/EGL/extensions/KHR/EGL_KHR_surfaceless_context.txt) is supported, it's possible to have `m_surface = EGL_NO_SURFACE`.

However the context is now without a default framebuffer, which will break apps that don't setup FBOs or readback from the default one. Keeping those functional is interesting in a CI context.

Reply:
> However the context is now without a default framebuffer, which will break apps that don't setup FBOs or readback from the default one. Keeping those functional is interesting in a CI context.

Actually that's good point, there should be some way to detect/validate that there is headless setup, and that user should not write into backbuffer.

Reply:
Some users are morons (based on years of experience with this project), so they will setup headless and then expect they see something... Therefore assert should give them a hint that headless means no backbuffer / default window.